src/GNNGraphs/transform.jl

```diff
@@ -188,6 +188,60 @@ function remove_multi_edges(g::GNNGraph{<:COO_T}; aggr = +)
              g.ndata, edata, g.gdata)
 end
 
+"""
+    remove_nodes(g::GNNGraph, nodes_to_remove::Vector{Int})
+
+Remove specified nodes, and their associated edges, from a GNNGraph.
+
+# Arguments
+- `g`: The input graph from which nodes (and their edges) will be removed.
+- `nodes_to_remove`: Vector of node indices to be removed.
+
+# Returns
+A new GNNGraph with the specified nodes and all edges associated with these nodes removed. 
+
+# Example
+```julia
+using GraphNeuralNetworks
+
+g = GNNGraph([1, 1, 2, 2, 3], [2, 3, 1, 3, 1])
+
+# Remove nodes with indices 2 and 3, for example
+g_new = remove_nodes(g, [2, 3])
+
+# g_new now does not contain nodes 2 and 3, and any edges that were connected to these nodes.
+println(g_new)
+```
+"""
+function remove_nodes(g::GNNGraph{<:COO_T}, nodes_to_remove)
```

Comment:
```suggestion
function remove_nodes(g::GNNGraph{<:COO_T}, nodes_to_remove::AbstractVector)
```
Could also add `nodes_to_remove = union(nodes_to_remove)` to handle duplicates.